autoload/airline/extensions/ale.vim

```diff
@@ -8,6 +8,28 @@ function! s:airline_ale_count(cnt, symbol)
   return a:cnt ? a:symbol. a:cnt : ''
 endfunction
 
+function! s:airline_ale_get_line_number(cnt, type) abort
+  if a:cnt == 0
+    return ''
+  endif
+
+  let buffer       = bufnr('')
+  let problem_type = (a:type ==# 'error') ? 'E' : 'W'
+  let problems     = copy(ale#engine#GetLoclist(buffer))
+
+  call filter(problems, 'v:val.bufnr =~ "' . buffer .
+                  \'" && v:val.type  =~ "' . problem_type . '"')
```

Comment:
You can write the variable names into the string instead. You should use `is` for exact matches, instead of `=~` which uses a regular expression match.

Reply:
@w0rp done. Check it it meets your standards.

Reply:
@w0rp could you give an example of what you mean by writing the variable name into the string?

Reply:
```vim
call filter(problems, 'v:val.bufnr is l:buffer && v:val.type is# l:problem_type')
```

Do that.

Reply:
Ah I see. I thought it would be interpreted as a string and not as a funcref. Thank you!